libcudacxx/include/cuda/std/__exception/exception_macros.h

```diff
@@ -84,8 +96,26 @@ _CCCL_END_NAMESPACE_CUDA_STD
     else                          \
     {                             \
     }
-#  define _CCCL_THROW(...) ::cuda::std::terminate()
-#  define _CCCL_RETHROW    ::cuda::std::terminate()
+#  define _CCCL_THROW(...)                                                                      \
+    do                                                                                          \
+    {                                                                                           \
+      NV_IF_TARGET(NV_IS_HOST, ({                                                               \
+                     ::cuda::__detail::__msg_storage __msg_buffer{};                            \
+                     ::cuda::std::__detail::__format_error(__msg_buffer, (__VA_ARGS__).what()); \
```

Comment:
Hm, it would be probably also nice to provide the type of the exception, something like: 
```suggestion
                     ::cuda::std::__detail::__format_error(__msg_buffer, typid(decltype(__VA_ARGS__)).name(), (__VA_ARGS__).what()); \
```